File: Algo-Review/02-BinarySearch/BinarySearch.java
public class BinarySearch {
    public static void main(String args[]){
        int[] array = {13, 24, 30, 35, 44, 48, 59, 67, 71, 82, 89, 91, 108, 159, 201};
        int target = 91;
        int index = binary_search(array, target);

        if( index == -1){
            System.out.println(target + " not found");
        }
        else{
            System.out.println("Element "+ target + " at index " + index);
        }
    }

    private static int binary_search(int[] array, int target){
        int low = 0;
        int high = array.length -1;

        while(low <= high){
            int middle = low + (high-low)/2;
            int value = array[middle];

            // if(value < target){
            //     low = middle + 1;
            // }
            // else if(value > target){
            //     high = middle - 1;
            // }
            // else{
            //     return middle;
            // }

            if(value < target) low = middle + 1;
            else if(value > target) high = middle - 1;
            else return middle;
            
        }
        return -1;
    }
}

File: Algo-Review/02-BinarySearch/BinarySearch.cpp
#include<stdio.h>
#define SIZE 15
int binarySearch(int array[], int target)
{
    int low = 0;
    int high = SIZE;
    while(low <= high){
        int middle = low + (high - low)/2;
        int value = array[middle];
        if(value <  target){
            low = middle + 1;
        }
        else if(value > target){
            high = middle - 1;
        }
        else{
            printf("middle: " + middle);
            return middle;
        }

    }

    return -1;
}

int main()
{
    int target = 91;
    int array[SIZE] = {13, 24, 30, 35, 44, 48, 59, 67, 71, 82, 89, 91, 108, 159, 201};
    int index = binarySearch(array, target);
    if ( index == -1){
        printf(target + " not found");
    } 
    else {
        printf("\nElement is at index " + index);
        printf("\n\n\n");
    }
}